lib/src/models/view/schedule_search.dart

```diff
@@ -18,11 +18,14 @@ class ScheduleSearchModel {
 
 	/// Finds all courses that match the given query.
 	/// 
-	/// The query parameter should be a part of a course's name, and classes will
-	/// be matched by searching against [Subject.name].
+	/// The query parameter should be a part of a course's name, id, or teacher
+	/// and classes will be matched by searching against
+	/// [Subject.name], [Subject.id], or [Subject.teacher].
 	List<Subject> getMatchingClasses(String query) => [
 		for (final Subject subject in subjects)
-			if (subject.name.contains(query))
+			if (subject.name.toLowerCase().contains(query.toLowerCase())||
+					subject.id.toLowerCase().contains(query.toLowerCase())||
+					subject.teacher.toLowerCase().contains(query.toLowerCase()))
 				subject
```

Comment:
```suggestion
			if (
				subject.name.toLowerCase().contains(query.toLowerCase())
				|| subject.id.toLowerCase().contains(query.toLowerCase())
				|| subject.teacher.toLowerCase().contains(query.toLowerCase())
			) subject

```

Small change, but looks a lot cleaner IMO

Reply:
Also, see the above comment about `.toLowerCase`